### ISSUE TITLE
function signature for itemFilter does not include `option` parameter causing typescript error

### ISSUE THREAD

**Issue by hughlaw** [rob-balfre/svelte-select#534]
If you are overriding the itemFilter logic by specifying a custom itemFilter prop on your select, Typescript will complain that the 3rd parameter that _does_ exist in the callback, does not exist.

The type definition for the itemFilter is currently: 
```
itemFilter?: (label: any, filterText: any) => boolean;
```
but this should be
```
itemFilter?: (label: any, filterText: any, option: any) => boolean;
```

**Comment by rob-balfre** [comment]
It's helpful to pass in the whole option incase people want to filter by a
different attribute to the label. I'll investigate how I can bypass the TS
error when I'm back from hols. Personally don't code with TS so might need
some pointers. Thanks

I'll probably just update the filter to use option[label] by default. That
would be non breaking and fix the TS issue I presume?

**Comment by hughlaw** [comment]
Yeah, that's pretty much how I discovered it - I'm passing in an object for each option and filtering based in multiple properties. Everything works, just typescript complains about it 😆

I forked the project and was going to submit a PR but couldn't figure out how the types are generated... is there a build process that create the types as part of the nom package? Happy to take a look if you can point me in the right direction.

**Comment by rob-balfre** [comment]
It's automatically done by the kit package process. If you update the
default filter to always use option it should fix the issue. Away from comp
though so cant help

**Comment by rob-balfre** [comment]
Should be fixed in 5.1.4

Thanks @hughlaw
